CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVCore.java

```diff
@@ -0,0 +1,205 @@
+    
+    Timer                       pollTimer = null;
+    ExtensionWebSocketClient    client  = null;
+    CSV                  csv    = null;
+    
+    final Logger log;
+    final static int RECONNECT_INTERVAL = 5000;
+
+    /**
+     * Stops sending messages to the source and tries to reconnect, closing on a failure
+     */
+    public final Handler<ExtensionServiceMessage> reconnectHandler = new Handler<ExtensionServiceMessage>() {
+        @Override
+        public void handleMessage(ExtensionServiceMessage message) {
+            log.trace("Reconnect message received. Reinitializing configuration");
+            
+            if (pollTimer != null) {
+                pollTimer.cancel();
+                pollTimer = null;
+            }
+            
+            oConfigHandler.configComplete = false;
+                        
+            CompletableFuture<Boolean> success = client.connectToSource();
+            
+            try {
+                if ( !success.get(10, TimeUnit.SECONDS) ) {
+                    if (!client.isOpen()) {
+                        log.error("Failed to connect to server url '" + targetVantiqServer + "'.");
+                    } else if (!client.isAuthed()) {
+                        log.error("Failed to authenticate within 10 seconds using the given authentication data.");
+                    } else {
+                        log.error("Failed to connect within 10 seconds");
+                    }
+                    close();
+                }
+            } catch (InterruptedException | ExecutionException | TimeoutException e) {
+                log.error("Could not reconnect to source within 10 seconds: ", e);
+                close();
+            }
+        }
+    };
+    
+    /**
+     * Stops sending messages to the source and tries to reconnect indefinitely
+     */
+    public final Handler<ExtensionWebSocketClient> closeHandler = new Handler<ExtensionWebSocketClient>() {
+        @Override
+        public void handleMessage(ExtensionWebSocketClient message) {
+            log.trace("WebSocket closed unexpectedly. Attempting to reconnect");
+            
+            if (pollTimer != null) {
+                pollTimer.cancel();
+                pollTimer = null;
+            }
+   
+            oConfigHandler.configComplete = false;
+            
+            boolean sourcesSucceeded = false;
+            while (!sourcesSucceeded) {
+                client.initiateFullConnection(targetVantiqServer, authToken);
+                sourcesSucceeded = exitIfConnectionFails(client, 10);
+                if (!sourcesSucceeded) {
+                    try {
+                        Thread.sleep(RECONNECT_INTERVAL);
+                    } catch (InterruptedException e) {
+                        log.error("An error occurred when trying to sleep the current thread. Error Message: ", e);
+                    }
+                }
+            }
+        }
+    };    
```

Comment:
extraneous semi-colon, I think.

```suggestion
    }    
```

Reply:
done

Reply:
its not a function , its field declaration . ; must stay .